### PR TITLE
Add "BlockParticipant" feature to `EOSLobbyManager.cs`

### PR DESCRIPTION
**Pull request by Xeynt**
Affected Class: EOSLobbyManager.cs
Added the following functions:
public void BlockTargetRTCParticipant(bool status, ProductUserId targetUserId, OnLobbyCallback blockRTCParticipantComplete) private void OnRTCBlockParticipantCompleted(ref BlockParticipantCallbackInfo data)

Modified the following (Class inside of EOSLobbyManager.cs): 
public class LobbyRTCState
Added: isBlocked variable

### CODE REVIEW

File: Assets/Scripts/EOSLobbyManager.cs
Comment (paulhazen) [inline Assets/Scripts/EOSLobbyManager.cs]: The _only_ thing that I believe needs to be changed before this can be approved is that (for the sake of consistency with our team's coding standards and for the other field members within `LobbyRTCState`) this field member should be capitalized since it is public (it should be changed to `IsBlocked`).

**Review by paulhazen — CHANGES_REQUESTED** [review]
Thanks for contributing @Ermelious! Fun to have more contributors. There is a very small change that I think is required before we can approve this and merge it into the `development` branch.

I took the liberty of simplifying the title of your PR for the sake of both clarity and brevity. I additionally added myself as an assignee, added the `enhancement` label, and requested reviews from the other members of the team. We require two team members to approve of a change before it can be merged.

When someone external to the team makes contributions, we have someone be co-assigned to it (in this case me) to champion the changes internally and help shepherd it toward approval.

Please follow up if you have any questions or concerns. I will similarly respond to your other PR later today.

File: Assets/Scripts/EOSLobbyManager.cs
Comment (andrew-hirata-playeveryware) [inline Assets/Scripts/EOSLobbyManager.cs]: A minor nit-pick on this: 
I would change this to `UpdateBlockStatusForRTCParticipant`, and flip the order of the parameters such that it's `('ProductUserId targetUserID, bool status, OnLobbyCallback blockRTCParticipantComplete)`

**Review by andrew-hirata-playeveryware — DISMISSED** [review]

**Comment by Xeynt** [comment]
> Thanks for contributing @Ermelious! Fun to have more contributors. There is a very small change that I think is required before we can approve this and merge it into the `development` branch.
> 
> I took the liberty of simplifying the title of your PR for the sake of both clarity and brevity. I additionally added myself as an assignee, added the `enhancement` label, and requested reviews from the other members of the team. We require two team members to approve of a change before it can be merged.
> 
> When someone external to the team makes contributions, we have someone be co-assigned to it (in this case me) to champion the changes internally and help shepherd it toward approval.
> 
> Please follow up if you have any questions or concerns. I will similarly respond to your other PR later today.

@paulhazen Happy to help. Yes, excuse the slip up, should have monitored the coding style of the other variables. Have pushed in the changes as requested. 

> A minor nit-pick on this: I would change this to `UpdateBlockStatusForRTCParticipant`, and flip the order of the parameters such that it's `('ProductUserId targetUserID, bool status, OnLobbyCallback blockRTCParticipantComplete)`

@andrew-hirata-playeveryware Sure, the changes has been made in the commit.

**Review by paulhazen — APPROVED** [review]

File: Assets/Scripts/EOSLobbyManager.cs
Comment (WispyMouse) [inline Assets/Scripts/EOSLobbyManager.cs:1740]: Should we run the BlockParticipantCallback here?

It's a bit tricky, because the result code is success. So what do we invoke to say that this error happened? Maybe it's fine not to invoke, if this isn't the current room? I'm not sure.

File: Assets/Scripts/EOSLobbyManager.cs
Comment (WispyMouse) [inline Assets/Scripts/EOSLobbyManager.cs]: As per comment, this looks like it finds myself. I think that if I follow this logic, whenever you block someone, it'll mute set *your own* isBlocked to data.Blocked, not the status of the person you blocked. Does it seem like that'd happen to you? If so, this should be adjusted to find `data.ParticipantId`.

**Review by WispyMouse — CHANGES_REQUESTED** [review]
Can you validate the behavior in `OnRTCBlockParticipantCompleted` that it sets the block status on the correct user?

File: Assets/Scripts/EOSLobbyManager.cs
Comment (WispyMouse) [inline Assets/Scripts/EOSLobbyManager.cs]: I think this is meant to be capitalized `IsBlocked`.

**Comment by Xeynt** [comment]
@WispyMouse 
> As per comment, this looks like it finds myself. I think that if I follow this logic, whenever you block someone, it'll mute set _your own_ isBlocked to data.Blocked, not the status of the person you blocked. Does it seem like that'd happen to you? If so, this should be adjusted to find `data.ParticipantId`.

> I think this is meant to be capitalized `IsBlocked`.

> Can you validate the behavior in `OnRTCBlockParticipantCompleted` that it sets the block status on the correct user?

Thanks for spotting those 2 out. I've checked and tested this time on my end after discarding other changes and there are no errors. And yes, the actual function is blocking the participant even before this change to the callback function.
A new commit has been added with the changes. 

> Should we run the BlockParticipantCallback here?
> 
> It's a bit tricky, because the result code is success. So what do we invoke to say that this error happened? Maybe it's fine not to invoke, if this isn't the current room? I'm not sure.

From my understanding, this check is to simply give a heads up to the devs that something is wrong. It doesn't stops it since the function already ran. These checks were referenced from the existing functions in this plugin:

EOSLobbyManager.cs
```c#
private void OnRTCRoomUpdateReceivingCompleted(ref UpdateReceivingCallbackInfo data)
{
    OnLobbyCallback ToggleMuteCallback = data.ClientData as OnLobbyCallback;

    //if (data == null)
    //{
    //    Debug.LogError("Lobbies (OnRTCRoomUpdateReceivingCompleted): UpdateSendingCallbackInfo data is null");
    //    ToggleMuteCallback?.Invoke(Result.InvalidState);
    //    return;
    //}

    if (data.ResultCode != Result.Success)
    {
        Debug.LogErrorFormat("Lobbies (OnRTCRoomUpdateReceivingCompleted): error code: {0}", data.ResultCode);
        ToggleMuteCallback?.Invoke(data.ResultCode);
        return;
    }

    Debug.LogFormat("Lobbies (OnRTCRoomUpdateReceivingCompleted): Updated receiving status successfully. LocalUserId={0} Room={1}, IsMuted={2}", data.LocalUserId, data.RoomName, data.AudioEnabled == false);

    // Ensure this update is for our room
    if (!CurrentLobby.RTCRoomName.Equals(data.RoomName, StringComparison.OrdinalIgnoreCase))
    {
        Debug.LogErrorFormat("Lobbies (OnRTCRoomUpdateReceivingCompleted): Incorrect Room! CurrentLobby.RTCRoomName={0} != data.RoomName", CurrentLobby.RTCRoomName, data.RoomName);
        return;
    }

    // Update should be for remote user
    if (EOSManager.Instance.GetProductUserId() != data.LocalUserId)
    {
        Debug.LogErrorFormat("Lobbies (OnRTCRoomUpdateReceivingCompleted): Incorrect call for local member.");
        return;
    }

    foreach(LobbyMember lobbyMember in CurrentLobby.Members)
    { 
        if(lobbyMember.ProductId != data.ParticipantId)
        {
            continue;
        }

        lobbyMember.RTCState.IsLocalMuted = data.AudioEnabled == false;
        lobbyMember.RTCState.MuteActionInProgress = false;

        Debug.LogFormat("Lobbies (OnRTCRoomUpdateReceivingCompleted): Cache updated for '{0}'", lobbyMember.ProductId);

        _Dirty = true;
        break;
    }
}
```

**Review by WispyMouse — APPROVED** [review]
Thanks for making those changes, I approve! When we get two of the team's staff to approve we can pull it in to development.

File: Assets/Scripts/EOSLobbyManager.cs
Comment (paulhazen) [reply Assets/Scripts/EOSLobbyManager.cs]: Changes as requested have been made - so I am marking this as resolving in order to merge.

**Pull request merged**
